src/main/java/net/backupcup/mcde/screen/RollBenchScreen.java

```diff
@@ -12,33 +11,44 @@
 import net.backupcup.mcde.screen.handler.RollBenchScreenHandler;
 import net.backupcup.mcde.screen.util.EnchantmentSlotsRenderer;
 import net.backupcup.mcde.screen.util.ScreenWithSlots;
-import net.backupcup.mcde.util.EnchantmentSlot.Choice;
-import net.backupcup.mcde.util.EnchantmentSlot.ChoiceWithLevel;
+import net.backupcup.mcde.screen.util.TextWrapUtils;
+import net.backupcup.mcde.screen.util.TexturePos;
+import net.backupcup.mcde.util.Choice;
 import net.backupcup.mcde.util.EnchantmentSlots;
 import net.backupcup.mcde.util.EnchantmentUtils;
-import net.backupcup.mcde.util.Slots;
+import net.backupcup.mcde.util.SlotPosition;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
 import net.minecraft.client.gui.screen.ingame.HandledScreen;
 import net.minecraft.client.render.GameRenderer;
 import net.minecraft.client.util.math.MatrixStack;
 import net.minecraft.entity.player.PlayerInventory;
 import net.minecraft.inventory.Inventory;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
 import net.minecraft.text.MutableText;
 import net.minecraft.text.Text;
 import net.minecraft.util.Formatting;
 import net.minecraft.util.Identifier;
+import net.minecraft.util.Pair;
 
+@Environment(EnvType.CLIENT)
 public class RollBenchScreen extends HandledScreen<RollBenchScreenHandler> implements ScreenWithSlots {
-    private Inventory inventory;
-
-    private static Pattern wrap = Pattern.compile("(\\b.{1,40})(?:\\s+|$)");
-
-    private Optional<Slots> opened = Optional.empty();
-
+    private static enum RerollItemSilouette {
+        LAPIS, SHARD;
```

Comment:
`SHARD` does not make it clear that this is referring to the `Echo Shard`. Consider `ECHO_SHARD`.